### PR TITLE
RDMs

### PR DESCRIPTION
**Pull request by lcyyork**
## Description
I would like to revamp the RDMs class so that it has two derived classes for spin-dependent and spin-free RDMs. Previous implementation has many flags to keep track of the available quantities, which is hard to maintain. Meanwhile, I would also like to add a function to directly return the spin-free RDMs in the CI_RDMs class to better suited with the new RDMs class. This PR is part of an effort to minimize the memory footprint of RDMs in Forte.

## User Notes
- [ ] Features added
- [ ] Changes to compilation (if any)

## Checklist
- [ ] Added/updated tests of new features and included a reference `output.ref` file
- [ ] Removed comments in code and input files
- [ ] Documented source code
- [ ] Checked for redundant headers
- [ ] Checked for consistency in the formatting of the output file
- [ ] Documented new features in the manual
- [ ] Ready to go!

### CODE REVIEW

**Comment by lcyyork** [comment]
@fevangelista Hi Francesco, I would like to make changes for the RDMs class. The new class is currently called RDMs_NEW and hopefully going to replace the RDMs class. Because this is a big change to Forte, I would like to check with you first and see if the new class looks OK to you.

**Review by fevangelista — APPROVED** [review]
This looks good. I have only one modification that I disagree with. I think that the `ncmo_` variable should be replaced, but I would use `nmo_` or `norb_` instead of `no_`. `no_` could be confused with number of occupied (orbitals). Please run all tests before committing. It would also be nice if you could add a comment regarding the changes made in the PR.

File: forte/base_classes/active_space_method.h [hunk 1]
```diff
@@ -295,9 +323,10 @@ std::unique_ptr<ActiveSpaceMethod> make_active_space_method(
     std::shared_ptr<MOSpaceInfo> mo_space_info, std::shared_ptr<ActiveSpaceIntegrals> as_ints,
     std::shared_ptr<ForteOptions> options);
 
-std::vector<RDMs> transition_rdms(std::shared_ptr<ActiveSpaceMethod> m1,
-                                  std::shared_ptr<ActiveSpaceMethod> m2,
-                                  std::vector<std::pair<size_t, size_t>>, int max_rdm_level);
+// std::vector<std::shared_ptr<RDMs>> transition_rdms(std::shared_ptr<ActiveSpaceMethod> m1,
```
[the remark this answers is not in the document]
Comment (lcyyork) [reply forte/base_classes/active_space_method.h:326]: I don't know. I just saw it is not implemented so I commented it out. The plan for this part is to implement the CASSI method.

**Comment by lcyyork** [comment]
All tests passed on my computer.

**Pull request merged**